### PR TITLE
Check Result Detail pt2

### PR DESCRIPTION
**Pull request by nelsonkopliku**
# Description

This PR introduces the detailed information about the execution of a specific check on a target (only expect statements on a target of type host)

Successful (passing/failing)

check-detail-success

Check error/Agent timeout

check-detail-error-and-timeout

What's next:

- detailed visualization for a cluster wide check (expect_same statements)
- add a result icon in the title as per figma mockup
- improve facts visualization for scalar values (should support plain strings/numbers/booleans besides lists/objects)
- improve detail page title (markdown ?)
- improve `<ListView />` component to allow custom spacing (to reduce the distance between expectations, values and fact names from the respective value )
- improve URL edge cases (what happens if a detail for a wrong checkID/targetType/targetName was requested? probably an error)
- cleanup legacy code

Going to create items in the backlog

## How was this tested?
Automated tests.

### CODE REVIEW

File: assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetail.jsx [hunk 1]
```diff
@@ -1,23 +1,42 @@
...
-          <ObjectTree data={executionData} />
-        </div>
-      </div>
+      <ExpectationsResults
+        isCheckError={isCheckError}
+        expectStatementsResults={expectStatementsResults}
+        errorMessage={message}
+      />
+      <ExpectedValues isCheckError={isCheckError} expectedValues={values} />
+      <GatheredFacts gatheredFacts={facts || []} />
```
Comment (dottorblaster) [inline assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetail.jsx:36]: You don't need this `||` since you used a default in the upper statement

File: assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetail.jsx [hunk 2]
```diff
@@ -1,23 +1,42 @@
...
-        Values
-      </div>
-      <div className="tn-cluster-details w-full my-4 mr-4 bg-white shadow rounded-lg px-8 py-4">
-        Facts
-        <div className="mt-3">
-          <ObjectTree data={executionData} />
-        </div>
-      </div>
+      <ExpectationsResults
+        isCheckError={isCheckError}
```
Comment (dottorblaster) [inline assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetail.jsx:31]: Maybe `isError` would be just enough?

Comment (dottorblaster) [reply assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetail.jsx:31]: (or just `error`)

File: assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetail.jsx [hunk 3]
```diff
@@ -1,23 +1,42 @@
...
+        expectStatementsResults={expectStatementsResults}
```
Comment (dottorblaster) [inline assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetail.jsx:32]:
```suggestion
        results={expectStatementsResults}
```

File: assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetail.jsx [hunk 4]
```diff
@@ -1,23 +1,42 @@
...
+        errorMessage={message}
+      />
+      <ExpectedValues isCheckError={isCheckError} expectedValues={values} />
```
Comment (dottorblaster) [inline assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetail.jsx:35]:
```suggestion
      <ExpectedValues isError={isCheckError} expectedValues={values} />
```

File: assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetailPage.jsx [hunk 5]
```diff
@@ -49,6 +49,10 @@ function CheckResultDetailPage() {
 
   const checkDescription = getCheckDescription(catalog, checkID);
 
+  const targetID = isTargetHost(targetType)
+    ? (clusterHosts.find(({ hostname }) => hostname === targetName) || {})?.id
+    : clusterID;
```
Comment (dottorblaster) [inline assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetailPage.jsx:54]: I think this could be easily refactored using lodash, too bad we still don't have lodash

File: assets/js/components/ExecutionResults/CheckResultDetail/GatheredFacts.jsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+
+import ListView from '@components/ListView';
+import ObjectTree from '@components/ObjectTree/ObjectTree';
```
Comment (dottorblaster) [inline assets/js/components/ExecutionResults/CheckResultDetail/GatheredFacts.jsx:4]:
```suggestion
import ObjectTree from '@components/ObjectTree';
```

File: assets/js/components/ExecutionResults/CheckResultDetail/GatheredFacts.jsx [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+    return {
+      title: name,
+      content: value,
+      render: (factValue) =>
+        isFactError ? (
+          <span className="text-red-500">{message}</span>
+        ) : (
+          <ObjectTree className="mt-3" data={factValue || {}} />
+        ),
```
Comment (dottorblaster) [inline assets/js/components/ExecutionResults/CheckResultDetail/GatheredFacts.jsx:18]:
```suggestion
      render: (factValue = {}) =>
        isFactError ? (
          <span className="text-red-500">{message}</span>
        ) : (
          <ObjectTree className="mt-3" data={factValue} />
        ),
```

I'm a bit fried at the moment, but would this work?

**Review by dottorblaster — CHANGES_REQUESTED** [review]
Nice! Little things to adjust :+1:

File: assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetail.jsx (hunk 1 above)
Comment (dottorblaster) [reply assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetail.jsx:36]: RIP

File: assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetail.jsx [hunk 8]
```diff
@@ -1,23 +1,43 @@
...
+  const {
+    expectation_evaluations = [],
+    values = [],
+    facts = [],
+    message,
+  } = checkResult;
+
+  const isError = isAgentCheckError(checkResult);
+
+  const expectStatementsResults = getExpectStatements(expectation_evaluations);
```
Comment (arbulu89) [inline assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetail.jsx:27]: Let's not use `Results` and use `Evaluation` to match with our code

File: assets/js/components/ExecutionResults/checksUtils.js [hunk 9]
```diff
@@ -125,3 +118,50 @@ export const getCheckExpectations = (catalog, checkID) => {
...
+
+export const isTargetHost = (targetType) => targetType === TARGET_HOST;
+export const isTargetCluster = (targetType) => targetType === TARGET_CLUSTER;
+
+export const isExpect = ({ type }) => type === EXPECT;
+export const isExpectSame = ({ type }) => type === EXPECT_SAME;
+
+export const isAgentCheckError = ({ type }) => !!type;
+
+export const getExpectStatements = (expectationList) =>
```
Comment (arbulu89) [inline assets/js/components/ExecutionResults/checksUtils.js:130]: I know that it is a bit redundant, but I would go with `getExpectExpectations`. 
This `Statement` word is coming like out of the blue (I know, already implemented in the prev PRs, but it striked me now)
Or maybe have `filterExpectationByType(type, expectationList)`

**Review by arbulu89 — COMMENTED** [review]
Hey @nelsonkopliku ,
The code looks great!
I mentioned some annoying naming things mostly, but besides that I generally like it

File: assets/js/lib/test-utils/factories/executions.js [hunk 10]
```diff
@@ -10,7 +10,7 @@ const expectationReturnTypeEnum = () =>
   faker.helpers.arrayElement(['expect', 'expect_same']);
 
 export const executionValueFactory = Factory.define(() => ({
-  name: faker.animal.cat(),
+  name: faker.lorem.word(),
```
[the remark this answers is not in the document]
Comment (nelsonkopliku) [reply assets/js/lib/test-utils/factories/executions.js]: Yes, that's exactly the issue :smile: changing to use sequence

File: assets/js/components/ExecutionResults/checksUtils.js [hunk 11]
```diff
@@ -125,3 +118,50 @@ export const getCheckExpectations = (catalog, checkID) => {
   }
   return [];
 };
+
+export const isTargetHost = (targetType) => targetType === TARGET_HOST;
```
[the remark this answers is not in the document]
Comment (nelsonkopliku) [reply assets/js/components/ExecutionResults/checksUtils.js:122]: I see, I will move them on top.
I was even thinking they can go somewhere in `lib/models`, but let's maybe agree together.

I am not sure I totally like what this checksUtils is becoming :sweat_smile:

File: assets/js/components/ExecutionResults/checksUtils.js [hunk 12]
```diff
@@ -125,3 +118,50 @@ export const getCheckExpectations = (catalog, checkID) => {
...
+  );
+
+  if (!expectSameStatement) {
+    return {};
+  }
+
+  return expectSameStatement;
+};
+
+export const getCheckResultByAgentID = (executionData, checkID, agentID) => {
```
[the remark this answers is not in the document]
Comment (nelsonkopliku) [reply assets/js/components/ExecutionResults/checksUtils.js:148]: what about singularized? `getAgentCheckResultByAgentID`?

File: assets/js/components/ExecutionResults/checksUtils.js (hunk 9 above)
Comment (arbulu89) [reply assets/js/components/ExecutionResults/checksUtils.js:130]: But we call them `expectations`, not `statements`. 
Anyhow, I'm fine with everyone else likes it

**Review by arbulu89 — APPROVED** [review]
Green light from my side

**Review by jagabomb — APPROVED** [review]
LGTM... I was going to mention a point about the spacing but I see you have already stated it in the description which I have referenced below. I'm excited to see the next set of improvements.

> improve <ListView /> component to allow custom spacing (to reduce the distance between expectations, values and fact names from the respective value )

**Review by dottorblaster — APPROVED** [review]
LGTM
